sdk/src/main/java/com/akkaserverless/javasdk/view/ViewBase.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright 2021 Lightbend Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.akkaserverless.javasdk.view;
+
+import com.akkaserverless.javasdk.impl.view.ViewUpdateEffectImpl;
+
+import java.util.Optional;
+
+/*
+ * FIXME rename this to View when the old annotation is removed
+ */
+/** @param <S> The type of the state for this view. */
+public abstract class ViewBase<S> {
+
+  private Optional<UpdateHandlerContext> updateHandlerContext = Optional.empty();
+
+  /**
+   * Additional context and metadata for an update handler.
+   *
+   * <p>It will throw an exception if accessed from constructor.
+   */
+  protected final UpdateHandlerContext updateHandlerContext() {
```

Comment:
We send commands to entities and therefore we have a `CommandContext`. For views, we could have an `UpdateContext`. 

Having `handler` in the name sound weird.

Reply:
UpdateContext sounds good to me